internal/pkg/cli/deploy/deploy.go

```diff
@@ -1210,8 +1210,16 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
 }
 
 func (d *backendSvcDeployer) validateALBRuntime() error {
-	if d.backendMft.RoutingRule.EmptyOrDisabled() || d.backendMft.RoutingRule.Alias.IsEmpty() || !d.env.HasImportedCerts() {
+	switch {
+	case d.backendMft.RoutingRule.Alias.IsEmpty() && d.env.HasImportedCerts():
+		return &errSvcWithNoALBAliasDeployingToEnvWithImportedCerts{
+			name:    d.name,
+			envName: d.env.Name,
+		}
+	case d.backendMft.RoutingRule.Alias.IsEmpty():
 		return nil
+	case !d.env.HasImportedCerts():
+		return fmt.Errorf(`cannot specify "alias" in an environment without imported certs`)
```

Comment:
What do you think of re-structuring like this:
```suggestion
	case d.backendMft.RoutingRule.Alias.IsEmpty() && !d.env.HasImportedCerts():
		return fmt.Errorf(`cannot specify "alias" in an environment without imported certs`)
	default:
		return nil
```

Reply:
problem is we want the case where the certs are imported and the alias is not empty to fall through to the checks below, but i'm happy to make the case more verbose like 
```suggestion
	case d.backendMft.RoutingRule.Alias.IsEmpty():
		return nil
	case !d.backendMft.RoutingRule.Alias.IsEmpty() && !d.env.HasImportedCerts():
		return fmt.Errorf(`cannot specify "alias" in an environment without imported certs`)
```
if you think that helps with readability😊